core/src/test/java/org/apache/iceberg/util/TestExceptionUtil.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.util;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class TestExceptionUtil {
+  private static final Logger LOG = LoggerFactory.getLogger(TestExceptionUtil.class);
+
+  private static class CustomCheckedException extends Exception {
+    private CustomCheckedException(String message) {
+      super(message);
+    }
+  }
+
+  @Test
+  public void testRunSafely() {
```

Comment:
I may be a bit paranoid, but I'd like to have a test which uses three exception types. Also a case for just testing runtime exceptions in each block.

Other than that I think this is a good utility! Thanks for working on it

Reply:
Done